service/history/queues/executable.go

```diff
@@ -320,6 +311,22 @@ func (e *executableImpl) Execute() (retErr error) {
 	return resp.ExecutionErr
 }
 
+func (e *executableImpl) writeToDLQ(ctx context.Context) error {
+	start := e.timeSource.Now()
+	err := e.dlqWriter.WriteTaskToDLQ(
+		ctx,
+		e.clusterMetadata.GetCurrentClusterName(),
+		e.clusterMetadata.GetCurrentClusterName(),
+		e.GetTask(),
+	)
+	if err != nil {
+		metrics.TaskDLQFailures.With(e.taggedMetricsHandler).Record(1)
+		e.logger.Error("Failed to write task to DLQ", tag.Error(err))
```

Comment:
Do we still want to emit TaskDLQSendLatency event when it fails to write a task to DLQ?

Reply:
Hmm, that's a good question. I think we still should, but I don't feel strongly either way. Let me know what you think

Reply:
I don't have a strong opinion. I am ok either way.